### PR TITLE
Introduce the Tuleap Authentication Jenkins plugin.

### PR DESCRIPTION
**Pull request by robinsoc**
This change is part of [community #14019](https://tuleap.net/plugins/tracker/?aid=14019)

This plugin does nothing for now. There is only a form which is displayed in the Security Realm menu.

 /!\ WARNING /!\: If you have selected "Tuleap Authentication" method, you should not save the configuration if you still want to connect to Jenkins.

    How to test:
     - Go to the 'Global Configuration Security' menu
    => You should see the "Tuleap Authentication" button radio in the "Security Realm" part
    => If you click on it you should see the form.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    <name>Tuleap Authentication Plugin</name>
+    <description>A Jenkins authentication plugin that delegates the authentication to a Tuleap server</description>
+
+    <licenses>
+        <license>
+            <name>MIT License</name>
+            <url>https://opensource.org/licenses/MIT</url>
+        </license>
+    </licenses>
+    <!-- Assuming you want to host on @jenkinsci:
```
Comment (Erwyn) [inline pom.xml]: can we remove this ?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    <name>Tuleap Authentication Plugin</name>
+    <description>A Jenkins authentication plugin that delegates the authentication to a Tuleap server</description>
+
+    <licenses>
+        <license>
+            <name>MIT License</name>
+            <url>https://opensource.org/licenses/MIT</url>
+        </license>
+    </licenses>
```
Comment (LeSuisse) [inline pom.xml]: It would be nice to have an explicit LICENSE file in the repo. It helps automated tooling to autodetect the license.

Comment (LeSuisse) [reply pom.xml]: Oh it's already there, forgot about it!

File: pom.xml (hunk 1 above)
Comment (robinsoc) [reply pom.xml]: done

**Pull request merged**